### PR TITLE
Introduction Text + Renaming Client Key Attestation to Client Attestation

### PR DESCRIPTION
**Pull request by paulbastian**

### CODE REVIEW

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 1]
```diff
@@ -55,20 +55,22 @@ informative:
...
+This specification defines a new method of client authentication for OAuth2 {{RFC6749}} by extending the approach defined in {{RFC7521}}. This new method enables client deployments that are traditionally viewed as public clients to be able to authenticate with the authorization server through an attestation based authentication scheme.
 
 --- middle
 
 # Introduction
 
-{{RFC7521}} defines a way for a client to include an assertion in a token request to an authorization server for the purposes of client authentication. This specification extends this mechanism to provide a way for a client instance to authenticate it self with the authorization server through an attested key based authentication scheme.
+{{RFC7521}} defines a way for a client to include an assertion in a token request to an authorization server for the purposes of client authentication. This specification extends this mechanism to provide a way for a client instance to authenticate it self with the authorization server through an attestation based authentication scheme.
```
Comment (tlodderstedt) [inline draft-looker-oauth-attested-key-based-client-authentication.md:64]:
```suggestion
{{RFC7521}} defines a way for a client to include an assertion in a token request to an authorization server for the purposes of client authentication. This specification extends this mechanism to provide a way for a client instance to authenticate itself with the authorization server through an assertion that is sender-constrained with a public key, i.e. the client must present the assertion along with a proof of possession for that public key. 
```

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 2]
```diff
@@ -55,20 +55,22 @@ informative:
...
...uthorization Server will need to obtain a Client Attestation JWT from its client backend first. Therefore the client will generate an Client Instance Key and possibly further attestations to proof its genuinity and security to the client backend. The Client instance sends this data to the client backend in request for a Client Attestation JWT. If the Client Backend successfully validates the Client Instance Key and further data, it will generate a signed Client Attestation JWT. As the Client Attestation JWT is cryptographically bound to the Client Instance Key generated by the client, the attestation is bound to this very specific client instance. The client backend will respond to the client's request by sending the Client Attestation JWT. The client can proceed and generate a Client Attestation Proof of Possession (PoP) for the Client Instance Key. Lastly, the client sends both the Client Attestation JWT and the Client Attestation PoP with the Token Request to the Authorization Server. The Authorization Server will validate the client attestation and thus authenticates the client instance. The Authorization Server may continue to issue sender-constrained access tokens using DPoP.
```
Comment (tlodderstedt) [inline draft-looker-oauth-attested-key-based-client-authentication.md:66]: what is an "attestation scheme"?

Comment (tlodderstedt) [inline draft-looker-oauth-attested-key-based-client-authentication.md:66]:
```suggestion
The Authorization Server will communicate its requirements for the attestation scheme through its metadata. A Client instance that wants to request an access token from this Authorization Server will need to obtain a Client Attestation JWT from its client backend first. Therefore the client will generate an ephemeral key (Client Instance Key) and (platform specific) attestations to proof its genuinity and security to the client backend. The Client instance sends this data to the client backend in request for a Client Attestation JWT. If the Client Backend successfully validates the Client Instance Key and further data, it will generate a signed Client Attestation JWT. As the Client Attestation JWT is cryptographically bound to the Client Instance Key generated by the client, the attestation is bound to this particular client instance. The client backend will respond to the client's request by sending the Client Attestation JWT. The client can proceed and generate a Client Attestation Proof of Possession (PoP) for the Client Instance Key. Lastly, the client sends both the Client Attestation JWT and the Client Attestation PoP with the Token Request to the Authorization Server. The Authorization Server will validate the client attestation and thus authenticates the client. The Authorization Server may continue to issue sender-constrained access tokens using DPoP.
```

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 3]
```diff
@@ -94,34 +96,45 @@ The following diagram depicts the conceptual interactions.
...
+2. A client attestation proof of possession (PoP) - produced by the client instance.
 
 # Conventions and Definitions
 
 {::boilerplate bcp14-tagged}
 
+# Terminology
+
+Client Attestation JWT
+   A Json Web Token (JWT) generated by the client backend that authenticates the client instance.
```
Comment (tlodderstedt) [inline draft-looker-oauth-attested-key-based-client-authentication.md:123]:
```suggestion
   A JSON Web Token (JWT) generated by the client backend that authenticates the client instance.
```

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 4]
```diff
@@ -243,7 +256,7 @@ Implementers should be aware that the design of this authentication mechanism de
...
+Authorization servers issuing a refresh token in response to a token request using the "urn:ietf:params:oauth:client-assertion-type:jwt-key-attestation" client authentication method MUST bind the refresh token to the client instance, and NOT just the client as specified in section 6 [@!RFC6749]. To prove this binding, the client instance MUST authenticate itself to the authorization server when refreshing an access token using the "urn:ietf:params:oauth:client-assertion-type:jwt-key-attestation" authentication method. The client MUST also use the same Client Instance Key that was used for authentication when the refresh token was issued.
```
Comment (tlodderstedt) [inline draft-looker-oauth-attested-key-based-client-authentication.md:259]: Why not just use the same client attestation JWT? That would not require any client instance identification and is also not a privacy issue. 

urn:ietf:params:oauth:client-assertion-type:jwt-key-attestation -> urn:ietf:params:oauth:client-assertion-type:jwt-client-attestation

**Review by tlodderstedt — CHANGES_REQUESTED** [review]

**Comment by tlodderstedt** [comment]
Please change file/draft name, too.

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 5]
```diff
@@ -94,34 +96,45 @@ The following diagram depicts the conceptual interactions.
...
                                  / \      |
                                   |       |
                                   +-------+
                                 (5) Generate
-                              Key Attestation PoP
+                           Client Attestation PoP
 ~~~
 
-Note defining the protocol for steps 2 and 4, including how the client instance authenticates with the client backend is out of scope of this specification.
+Note that the communciation protocols used for requesting and sending a Client Attestation JWT between the client instance and its backend (steps 2 and 4) and the data how the client instance authenticates with the client backend is out of scope of this specification.
```
Comment (tplooker) [inline draft-looker-oauth-attested-key-based-client-authentication.md:109]: This text feels a bit more complex to me? I think the previous text was a bit clearer in this explanation?

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 6]
```diff
@@ -94,34 +96,45 @@ The following diagram depicts the conceptual interactions.
...
 
-This specification only defines the format of the client assertion that a client instance uses to authenticate in its interactions with an authorization server (indicated in step 6), which is comprised of two key parts:
+This specification only defines the format and minimum schema of the Client Assertion JWT that a client instance uses to authenticate in its interactions with an authorization server (indicated in step 6), which is comprised of two key parts:
```
Comment (tplooker) [inline draft-looker-oauth-attested-key-based-client-authentication.md:111]: schema is a bit of a loaded word here I think the JWT processing section already makes it clear AS and clients are permitted to add what ever else they wish to the JWTs
```suggestion
This specification only defines the format of the Client Assertion JWT that a client instance uses to authenticate in its interactions with an authorization server (indicated in step 6), which is comprised of two key parts:
```

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 7]
```diff
@@ -94,34 +96,45 @@ The following diagram depicts the conceptual interactions.
...
 
-1. A client key attestation - produced by the client backend.
-2. A client key attestation proof of possession (PoP) - produced by the client instance.
+1. A client attestation Json Web Token (JWT)- produced by the client backend.
```
Comment (tplooker) [inline draft-looker-oauth-attested-key-based-client-authentication.md:113]:
```suggestion
1. A client attestation JSON Web Token (JWT)- produced by the client backend.
```

File: draft-looker-oauth-attested-key-based-client-authentication.md (hunk 2 above)
Comment (peppelinux) [reply draft-looker-oauth-attested-key-based-client-authentication.md:66]: A well know data structure that defines the data an attestation should, shall or may, contain

Should It be a defined term?

**Review by tlodderstedt — CHANGES_REQUESTED** [review]

File: draft-looker-oauth-attested-key-based-client-authentication.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (paulbastian) [reply draft-looker-oauth-attested-key-based-client-authentication.md]: outdated, but approved the change in my commit

**Review by tlodderstedt — APPROVED** [review]

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 8]
```diff
@@ -55,20 +55,22 @@ informative:
...
...s Authorization Server will need to obtain a Client Attestation JWT from its client backend first. Therefore the client will generate an key (Client Instance Key) and (platform specific) attestations to proof its genuinity and security to the client backend. The Client instance sends this data to the client backend in request for a Client Attestation JWT. If the Client Backend successfully validates the Client Instance Key and further data, it will generate a signed Client Attestation JWT. As the Client Attestation JWT is cryptographically bound to the Client Instance Key generated by the client, the attestation is bound to this particular client instance. The client backend will respond to the client's request by sending the Client Attestation JWT. The client can proceed and generate a Client Attestation Proof of Possession (PoP) for the Client Instance Key. Lastly, the client sends both the Client Attestation JWT and the Client Attestation PoP with the Token Request to the Authorization Server. The Authorization Server will validate the client attestation and thus authenticates the client. The Authorization Server may continue to issue sender-constrained access tokens using DPoP.
```
Comment (Sakurann) [inline draft-looker-oauth-attested-key-based-client-authentication.md:66]:
```suggestion
The Authorization Server will communicate its requirements for the attestation scheme through its metadata. A Client instance that wants to request an access token from this Authorization Server will need to obtain a Client Attestation JWT, for example, from its client backend first. Therefore, the client will generate a key (Client Instance Key) and (platform specific) attestations to prove its authenticity and security to the client backend. The Client instance sends this data to the client backend in request for a Client Attestation JWT. If the Client Backend successfully validates the Client Instance Key and further data, it will generate a signed Client Attestation JWT. As the Client Attestation JWT is cryptographically bound to the Client Instance Key generated by the client, the attestation is bound to this particular client instance. The client backend will respond to the client's request by sending the Client Attestation JWT. The client can proceed and generate a Client Attestation Proof of Possession (PoP) for the Client Instance Key. Lastly, the client sends both the Client Attestation JWT and the Client Attestation PoP with the Token Request to the Authorization Server. The Authorization Server will validate the client attestation and thus authenticates the client. The Authorization Server may continue to issue sender-constrained access tokens using {{oauth-dpop-16}}.
```

what is `(platform specific) attestation` ?
please be consistent, capitalize `Client` everywhere or not capitalize it.
the explanation is pretty long. I would suggest to put a short summary before the diagram but break down this paragraph into the steps and put them under the diagram.

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 9]
```diff
@@ -94,34 +96,45 @@ The following diagram depicts the conceptual interactions.
...
+# Terminology
+
+Client Attestation JWT
+   A JSON Web Token (JWT) generated by the client backend that authenticates the client instance.
+
+Client Attestation Proof of Possession (PoP)
+   A Proof of Possession generated by the client instance for the key that is used in the Client Attestation JWT.
+
+Client Instance Key
+   A cryptographic, asymmetric key generated by the client instance and proven to the client backend. The public key is contained in the Client Attestation JWT and is used to sign the Client Attestation Proof of Possession
```
Comment (Sakurann) [inline draft-looker-oauth-attested-key-based-client-authentication.md:129]:
```suggestion
   A cryptographic, asymmetric key generated by the client instance and proven to the client backend. The public key is contained in the Client Attestation JWT and is used to sign the Client Attestation Proof of Possession.
```

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 10]
```diff
@@ -196,9 +209,9 @@ The following example is the decoded header and payload of a JWT meeting the pro
 }
 ~~~
 
-### Client Key Attestation PoP JWT {#client-key-attestation-pop-jwt}
+### Client Attestation PoP JWT {#client-attestation-pop-jwt}
 
-The following rules apply to validating the client key attestation PoP JWT. Application of additional restrictions and policy are at the discretion of the authorization server.
+The following rules apply to validating the client attestation PoP JWT. Application of additional restrictions and policy are at the discretion of the authorization server.
```
Comment (Sakurann) [inline draft-looker-oauth-attested-key-based-client-authentication.md:214]:
```suggestion
The following rules apply to validating the Client Attestation JWT. Application of additional restrictions and policy are at the discretion of the authorization server.
```
Client Attestation JWT or Client Attestation PoP JWT?
also capitalization - please be consistent...

**Review by Sakurann — CHANGES_REQUESTED** [review]
a new paragraph in intro sectionis great but hard to follow, suggest to break it down into steps and put under the diagram.

I also perceive this draft as a PoP version of urn:ietf:params:oauth:grant-type:jwt-bearer, would it make sense to make that more intuitive and explicit - like urn:ietf:params:oauth:grant-type:jwt-pop? - this can be another issue

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 11]
```diff
@@ -55,20 +55,22 @@ informative:
 
 --- abstract
 
-This specification defines a new method of client authentication for OAuth2 {{RFC6749}} by extending the approach defined in {{RFC7521}}. This new method enables client deployments that are traditionally viewed as public clients to be able to authenticate with the authorization server through an attested key based authentication scheme.
+This specification defines a new method of client authentication for OAuth 2.0 {{RFC6749}} by extending the approach defined in {{RFC7521}}. This new method enables client deployments that are traditionally viewed as public clients to be able to authenticate with the authorization server through an attestation based authentication scheme.
```
Comment (tplooker) [inline draft-looker-oauth-attested-key-based-client-authentication.md:58]: Just noting that given we now have OAuth 2.1 saying this is for OAuth 2.0 might give the wrong compatibility impression?

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 12]
```diff
@@ -55,20 +55,22 @@ informative:
...
 
 --- middle
 
 # Introduction
 
-{{RFC7521}} defines a way for a client to include an assertion in a token request to an authorization server for the purposes of client authentication. This specification extends this mechanism to provide a way for a client instance to authenticate it self with the authorization server through an attested key based authentication scheme.
+{{RFC7521}} defines a way for a client to include an assertion in a token request to an authorization server for the purposes of client authentication. This specification extends this mechanism to provide a way for a client instance to authenticate itself with the authorization server through assertion that is bound to a public key (for proof of possession). This assertion is designated as Client Attestation.
```
[the remark this answers is not in the document]
Comment (tplooker) [reply draft-looker-oauth-attested-key-based-client-authentication.md:64]:
```suggestion
{{RFC7521}} defines a way for a client to include an assertion in a token request to an authorization server for the purposes of client authentication. This specification uses this framework to define a new assertion type that provides a way for a client instance to authenticate itself with the authorization server through an assertion that is bound to a public key (for proof of possession). This assertion is designated with the name of Client Attestation in this draft.
```

File: draft-looker-oauth-attested-key-based-client-authentication.md (hunk 11 above)
Comment (paulbastian) [reply draft-looker-oauth-attested-key-based-client-authentication.md:58]: @tlodderstedt originally made this change

File: draft-looker-oauth-attested-key-based-client-authentication.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (tplooker) [reply draft-looker-oauth-attested-key-based-client-authentication.md:64]: Kind of conflicts with @Sakurann review, we don't extend the assertion we extend the framework or mechanism defined by RFC 7521.

File: draft-looker-oauth-attested-key-based-client-authentication.md (hunk 8 above)
Comment (tplooker) [reply draft-looker-oauth-attested-key-based-client-authentication.md:66]: Communication of this requirement may be communicated via metadata I dont think that is the only way though and would prefer we dont imply this. I'd also prefer we dont talk much about how the attestation is obtained by the client instance, much of this paragraph to me is informational and might set the wrong tone for the draft in terms of prescribing a solution.

Comment (paulbastian) [reply draft-looker-oauth-attested-key-based-client-authentication.md:66]: Client Attestation JWT is from the terminology, so I capitalize it, other occurences are not in the Terminology (yet), so therefore not capitalized, that should be consistent for now

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 13]
```diff
@@ -94,34 +96,45 @@ The following diagram depicts the conceptual interactions.
...
                                  / \      |
                                   |       |
                                   +-------+
                                 (5) Generate
-                              Key Attestation PoP
+                           Client Attestation PoP
 ~~~
 
-Note defining the protocol for steps 2 and 4, including how the client instance authenticates with the client backend is out of scope of this specification.
+Note that the data and protocols used for requesting and sending a Client Attestation JWT (steps 2 and 4) and how the client instance authenticates with the client backend is out of scope of this specification.
```
Comment (tplooker) [inline draft-looker-oauth-attested-key-based-client-authentication.md:109]: Still prefer the previous language to be honest, data exchanged is a part of the protocol IMO and its one protocol not multiple hence protocol instead of protocols. The diagram also already describes what is happening so the "request and sending a client attestation JWT" feels redundant.

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 14]
```diff
@@ -94,34 +96,45 @@ The following diagram depicts the conceptual interactions.
...
+
+Client Instance Key
+   A cryptographic, asymmetric key generated by the client instance and proven to the client backend. The public key is contained in the Client Attestation JWT and is used to sign the Client Attestation Proof of Possession
+
 # Client Authentication in the Token Request
 
 To perform client authentication using this scheme, the client instance uses the following parameter values and encodings.
 
-The value of the "client_assertion_type" is "urn:ietf:params:oauth:client-assertion-type:jwt-key-attestation".
+The value of the "client_assertion_type" is "urn:ietf:params:oauth:client-assertion-type:jwt-client-attestation".
```
[the remark this answers is not in the document]
Comment (tplooker) [reply draft-looker-oauth-attested-key-based-client-authentication.md:135]: Im not against this but I think it makes it less clear how it relates to the terminology we have established in the draft e.g this means the type would be ^ but the name of is client attestation?

File: draft-looker-oauth-attested-key-based-client-authentication.md (hunk 11 above)
Comment (tlodderstedt) [reply draft-looker-oauth-attested-key-based-client-authentication.md:58]: we don't have OAuth 2.1 yet, it's a draft. But I would be fine with just "OAuth" (with the existing reference to RFC 6749

File: draft-looker-oauth-attested-key-based-client-authentication.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (paulbastian) [reply draft-looker-oauth-attested-key-based-client-authentication.md:64]: I agree with @tplooker and like the emphasis on client instance

File: draft-looker-oauth-attested-key-based-client-authentication.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (paulbastian) [reply draft-looker-oauth-attested-key-based-client-authentication.md:135]: I agree with @tplooker , as that could lead to confusion. But no hard feelings here

**Comment by paulbastian** [comment]
> 
> I also perceive this draft as a PoP version of urn:ietf:params:oauth:grant-type:jwt-bearer, would it make sense to make that more intuitive and explicit - like urn:ietf:params:oauth:grant-type:jwt-pop? - this can be another issue

Opened as a separate issue.

**Comment by paulbastian** [comment]
> a new paragraph in intro sectionis great but hard to follow, suggest to break it down into steps and put under the diagram.

@Sakurann please check my latest suggestion

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 15]
```diff
@@ -55,20 +55,20 @@ informative:
 
 --- abstract
 
-This specification defines a new method of client authentication for OAuth2 {{RFC6749}} by extending the approach defined in {{RFC7521}}. This new method enables client deployments that are traditionally viewed as public clients to be able to authenticate with the authorization server through an attested key based authentication scheme.
+This specification defines a new method of client authentication for OAuth {{RFC6749}} by extending the approach defined in {{RFC7521}}. This new method enables client deployments that are traditionally viewed as public clients to be able to authenticate with the authorization server through an attestation based authentication scheme.
```
Comment (peppelinux) [inline draft-looker-oauth-attested-key-based-client-authentication.md:58]: does the status of "public clients" be applied also in this case?

I've the perception that in our use cases whatever are the statuses of the clients is not relevant to this draft

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 16]
```diff
@@ -94,34 +94,51 @@ The following diagram depicts the conceptual interactions.
...
                                  / \      |
                                   |       |
                                   +-------+
                                 (5) Generate
-                              Key Attestation PoP
+                           Client Attestation PoP
 ~~~
 
-Note defining the protocol for steps 2 and 4, including how the client instance authenticates with the client backend is out of scope of this specification.
+At first, the Authorization Server will communicate its requirements for the client authentication through its metadata. A Client instance that wants to request an access token from this Authorization Server will need to obtain a Client Attestation JWT, for example, from its client backend first.
```
Comment (peppelinux) [inline draft-looker-oauth-attested-key-based-client-authentication.md:107]: I'd write using present, removing "will"

WDYT about

`The flow starts with the AS communicating its requirements for the client authentication to the client, made available in its metadata.`

Having said "the AS" that would be the same for the Client as well, thus

`The Client instance makes available its Client Attestation JWS to the AS where an access token is requested`

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 17]
```diff
@@ -94,34 +94,51 @@ The following diagram depicts the conceptual interactions.
...
 
-This specification only defines the format of the client assertion that a client instance uses to authenticate in its interactions with an authorization server (indicated in step 6), which is comprised of two key parts:
+Therefore, the client will generate a key (Client Instance Key) and (platform specific) attestations to prove its authenticity and security to the client backend (step 1). The Client instance sends this data to the client backend in request for a Client Attestation JWT (step 2). If the Client Backend successfully validates the Client Instance Key and further data, it will generate a signed Client Attestation JWT (step 3). As the Client Attestation JWT is cryptographically bound to the Client Instance Key generated by the client, the attestation is bound to this particular client instance. The client backend will respond to the client's request by sending the Client Attestation JWT (step 4).
```
Comment (peppelinux) [inline draft-looker-oauth-attested-key-based-client-authentication.md:109]: I'd not use the future, I think that would be better talking in the present

Comment (peppelinux) [reply draft-looker-oauth-attested-key-based-client-authentication.md:109]: `client instance` -> `Client instance`

Comment (peppelinux) [reply draft-looker-oauth-attested-key-based-client-authentication.md:109]: is client backend a defined term? If yes, we should uppercase it accordingly

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 18]
```diff
@@ -94,34 +94,51 @@ The following diagram depicts the conceptual interactions.
...
 
-1. A client key attestation - produced by the client backend.
-2. A client key attestation proof of possession (PoP) - produced by the client instance.
+The client can proceed and generate a Client Attestation Proof of Possession (PoP) for the Client Instance Key (step 5). Lastly, the client sends both the Client Attestation JWT and the Client Attestation PoP with the Token Request to the Authorization Server. The Authorization Server will validate the client attestation and thus authenticates the client. The Authorization Server may continue to issue sender-constrained access tokens using {{oauth-dpop-16}}.
```
Comment (peppelinux) [inline draft-looker-oauth-attested-key-based-client-authentication.md:111]: will validate -> valitades

**Review by peppelinux — CHANGES_REQUESTED** [review]

File: draft-looker-oauth-attested-key-based-client-authentication.md (hunk 15 above)
Comment (paulbastian) [reply draft-looker-oauth-attested-key-based-client-authentication.md:58]: the clients being authenticated by the approach of this RFC, are to be considered public clients in OAuth speak. Doesn't it make sense to say so?

File: draft-looker-oauth-attested-key-based-client-authentication.md (hunk 17 above)
Comment (paulbastian) [reply draft-looker-oauth-attested-key-based-client-authentication.md:109]: I will capitalize things later once they have an entry in the terminology

File: draft-looker-oauth-attested-key-based-client-authentication.md (hunk 15 above)
Comment (peppelinux) [reply draft-looker-oauth-attested-key-based-client-authentication.md:58]: I'm ok with public client since wallet instances are mobile devices, I was just asking since I read in the specs also the role of client backend without knowing which would be their client registration mechanisms

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 19]
```diff
@@ -94,34 +94,51 @@ The following diagram depicts the conceptual interactions.
...
+
+Client Attestation JWT
+   A JSON Web Token (JWT) generated by the client backend that authenticates the client instance.
+
+Client Attestation Proof of Possession (PoP)
+   A Proof of Possession generated by the client instance for the key that is used in the Client Attestation JWT.
+
+Client Instance Key
+   A cryptographic, asymmetric key generated by the client instance and proven to the client backend. The public key is contained in the Client Attestation JWT and is used to sign the Client Attestation Proof of Possession.
+
```
Comment (peppelinux) [inline draft-looker-oauth-attested-key-based-client-authentication.md:134]: I approve this PR with this in mind

Client Instance, as represented in the Figure 1, should be a defined term as well

**Review by peppelinux — APPROVED** [review]

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 20]
```diff
@@ -94,34 +94,51 @@ The following diagram depicts the conceptual interactions.
...
                                  / \      |
                                   |       |
                                   +-------+
                                 (5) Generate
-                              Key Attestation PoP
+                           Client Attestation PoP
 ~~~
 
-Note defining the protocol for steps 2 and 4, including how the client instance authenticates with the client backend is out of scope of this specification.
+The flow starts with the Authorization Server communicating its requirements for the client authentication to the client, made available in its metadata. A Client instance that wants to request an access token from this Authorization Server needs to obtain a Client Attestation JWT, for example, from its client backend first.
```
Comment (tplooker) [inline draft-looker-oauth-attested-key-based-client-authentication.md:107]: This paragraph reads like there is a dependency on an AS providing a metadata document for a client to discover when in reality a client could know through an out of band mechanism that its what the AS requires for client authentication

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 21]
```diff
@@ -94,34 +94,51 @@ The following diagram depicts the conceptual interactions.
...
+2. A Client Attestation Proof of Possession (PoP) - produced by the client instance.
 
 # Conventions and Definitions
 
 {::boilerplate bcp14-tagged}
 
+# Terminology
+
+Client Attestation JWT
+   A JSON Web Token (JWT) generated by the client backend that authenticates the client instance.
```
Comment (tplooker) [inline draft-looker-oauth-attested-key-based-client-authentication.md:127]:
```suggestion
   A JSON Web Token (JWT) generated by the client backend which is bound to a key managed by a client instance which can then be used by the instance for client authentication.
```

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 22]
```diff
@@ -94,34 +94,51 @@ The following diagram depicts the conceptual interactions.
...
+
+Client Attestation Proof of Possession (PoP)
```
Comment (tplooker) [inline draft-looker-oauth-attested-key-based-client-authentication.md:129]:
```suggestion
Client Attestation Proof of Possession (PoP) JWT
```

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 23]
```diff
@@ -94,34 +94,51 @@ The following diagram depicts the conceptual interactions.
...
+   A Proof of Possession generated by the client instance for the key that is used in the Client Attestation JWT.
```
Comment (tplooker) [inline draft-looker-oauth-attested-key-based-client-authentication.md:130]:
```suggestion
   A Proof of Possession generated by the client instance using the key that the Client Attestation JWT is bound to.
```

**Review by Sakurann — COMMENTED** [review]
I would like to see the description under the sequence diagram improved, but if you want to address that in another PR, that is fine too.
